projects/storefrontlib/cms-components/navigation/scroll-to-top/scroll-to-top.component.spec.ts

```diff
@@ -43,86 +21,119 @@ const mockData: CmsScrollToTopComponent = {
 const MockCmsComponentData = <CmsComponentData<any>>{
   data$: of(mockData),
 };
+//TODO: (CXSPA-6522) - remove feature flag next major release.
+class MockFeatureConfigService {
+  isEnabled() {
+    return true;
+  }
+}
 
 describe('ScrollToTopComponent', () => {
-  let component: MockComponent;
-  let fixture: ComponentFixture<MockComponent>;
-  let winRef: WindowRef;
+  let component: ScrollToTopComponent;
+  let fixture: ComponentFixture<ScrollToTopComponent>;
   let focusUtility: SelectFocusUtility;
   let el: DebugElement;
+  let scrollBtn: HTMLElement;
 
   beforeEach(() => {
     TestBed.configureTestingModule({
       imports: [IconTestingModule, I18nTestingModule],
-      declarations: [MockComponent, ScrollToTopComponent],
+      declarations: [ScrollToTopComponent],
       providers: [
         {
           provide: CmsComponentData,
           useValue: MockCmsComponentData,
         },
-        { provide: WindowRef, useClass: MockWinRef },
-        { provide: SelectFocusUtility, useClass: MockSelectFocusUtility },
+        {
+          provide: FeatureConfigService,
+          useClass: MockFeatureConfigService,
+        },
       ],
     }).compileComponents();
 
     focusUtility = TestBed.inject(SelectFocusUtility);
-    winRef = TestBed.inject(WindowRef);
-    fixture = TestBed.createComponent(MockComponent);
+    fixture = TestBed.createComponent(ScrollToTopComponent);
 
     component = fixture.componentInstance;
     el = fixture.debugElement;
+    scrollBtn = el.query(By.css('.cx-scroll-to-top-btn')).nativeElement;
+    component.button = el.query(By.css('.cx-scroll-to-top-btn'));
   });
 
   it('should create component', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should not be displayed at top of page', () => {
-    fixture.detectChanges();
-    winRef.nativeWindow?.scrollTo(0, 0);
-    fixture.detectChanges();
-    const scrollComponent = el.query(By.css('.display'));
-    expect(scrollComponent).toBeNull();
+  it('should set config on init', () => {
+    spyOn<any>(component, 'setConfig').and.callThrough();
+    component.ngOnInit();
+    expect(component['setConfig']).toHaveBeenCalled();
   });
 
-  it('should be visible and scroll to top of page', () => {
-    fixture.detectChanges();
-    spyOn(focusUtility, 'findFirstFocusable').and.callThrough();
-    spyOn(window, 'scrollTo').and.callThrough();
-    winRef.nativeWindow?.scrollTo(0, 200);
-    winRef.nativeWindow?.dispatchEvent(new Event('scroll'));
+  it('should scroll window to top when clicked', () => {
+    spyOn<any>(component['window'], 'scrollTo');
+
+    component.scrollToTop(new MouseEvent('click'));
+    expect(component['window']?.scrollTo).toHaveBeenCalled();
```

Comment:
Bit of a nitpick, but should we expect a value that `scrollTo` was called with? ie. zero for top?